pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -2097,6 +2089,44 @@ pub mod pallet {
 			});
 		}
 
+		fn try_remove_avatar_ownership_from(
+			player: &AccountIdFor<T>,
+			season_id: &SeasonId,
+			avatar_id: &AvatarIdOf<T>,
+		) -> DispatchResult {
+			Owners::<T>::mutate(player, season_id, |avatars| {
+				avatars.retain(|id| id != avatar_id);
+			});
+
+			Avatars::<T>::try_mutate(avatar_id, |maybe_avatar| -> DispatchResult {
+				let (from_owner, _) = maybe_avatar.as_mut().ok_or(Error::<T>::UnknownAvatar)?;
+				*from_owner = Self::technical_account_id();
+				Ok(())
+			})
+		}
+
+		fn try_restore_avatar_ownership_to(
+			player: &AccountIdFor<T>,
+			season_id: &SeasonId,
+			avatar_id: &AvatarIdOf<T>,
+		) -> DispatchResult {
+			Owners::<T>::try_mutate(player, season_id, |avatars| {
+				avatars.try_push(*avatar_id).map_err(|_| Error::<T>::MaxOwnershipReached)?;
+				ensure!(
+					avatars.len() <=
+						PlayerSeasonConfigs::<T>::get(player, season_id).storage_tier as usize,
+					Error::<T>::MaxOwnershipReached
+				);
+				Ok::<_, DispatchError>(())
+			})?;
+
+			Avatars::<T>::try_mutate(avatar_id, |maybe_avatar| -> DispatchResult {
+				let (from_owner, _) = maybe_avatar.as_mut().ok_or(Error::<T>::UnknownAvatar)?;
+				*from_owner = player.clone();
```

Comment:
We don't want to check the current here? Shouldn't this only be used if the current owner is the technical account?

Reply:
We already check in `unlock_avatar` if the avatar is owned by the technical account, so the check here is not necessary.